Common++/header/IpAddress.h

```diff
@@ -566,6 +585,108 @@ namespace pcpp
+		 * @param netmask A string representing a netmask in valid IPv6 format, for example: ffff:ffff::.
+		 * Please notice that netmasks that start with zeros are invalid, for example: 0:ffff::. The only netmask
+		 * starting with zeros that is valid is all zeros (::). If the netmask is invalid std::invalid_argument
+		 * exception is thrown
+		 */
+		IPv6Network(const IPv6Address& address, const std::string& netmask);
+
+		/**
+		 * A constructor that creates an instance of the class out of a string representing the network prefix and
+		 * a prefix length or a netmask
+		 * @param addressAndNetmask A string in one of these formats:
+		 *  - IPV6_ADDRESS/Y where IPV6_ADDRESS is a valid IPv6 address representing the network prefix and Y is
+		 *    a number between 0 and 128 representing the network prefix
+		 *  - IPV6_ADDRESS/IPV6_NETMASK where IPV6_ADDRESS is a valid IPv6 address representing the network prefix
+		 *    and IPV6_NETMASK is a valid IPv6 netmask
+		 *  For any invalid value std::invalid_argument is thrown
+		 */
+		IPv6Network(const std::string& addressAndNetmask);
+
+		/**
+		 * @return The prefix length, for example: the prefix length of 3546::/ffff:: is 16
+		 */
+		uint8_t getPrefixLen() const;
+
+		/**
+		 * @return The netmask, for example: the netmask of 3546::/16 is ffff::
+		 */
+		std::string getNetmask() const { return IPv6Address(m_Mask).toString(); }
+
+		/**
+		 * @return The network prefix, for example: the network prefix of 3546:f321::/16 is 3546::
+		*/
+		IPv6Address getNetworkPrefix() const { return IPv6Address(m_NetworkPrefix); }
+
+		/**
+		* @return The lowest non-reserved IPv6 address in this network, for example: the lowest address in 3546::/16 is
+		* 3546::1
+		*/
+		IPv6Address getLowestAddress() const;
+
+		/**
+		 * @return The highest IPv6 address in this network, for example: the highest address in 3546::/16 is
+		 * 3546:ffff:ffff:ffff:ffff:ffff:ffff:ffff
+		 */
+		IPv6Address getHighestAddress() const;
+
+		/**
+		 * @return The number of addresses in this network, for example: the number of addresses in 16ff::/120 is 256.
+		 * If the number of addresses exceeds the size of uint64_t a std::out_of_range exception is thrown
+		 */
+		uint64_t getTotalAddressCount() const;
+
+		/**
+		 * @param address An IPv6 address
+		 * @return True is the address belongs to the network, false otherwise or if the address isn't valid
+		 */
+		bool includes(const IPv6Address& address) const;
+
+		/**
+		 * @param network An IPv6 network
+		 * @return True is the input network is completely included within this network, false otherwise, for example:
+		 * 3546::/64 includes 3546::/120 but doesn't include 3546::/16
+		 */
+		bool includes(const IPv6Network& network) const;
+
+		/**
+		 * @return A string representation of the network in a format of NETWORK_PREFIX/PREFIX_LEN, for example:
+		 * fda7:9f81:6c23:275::/64
+		 */
+		std::string toString() const;
+
+	private:
+		uint8_t m_NetworkPrefix[16];
```

Comment:
uint8_t m_NetworkPrefix[IPV6_ADDR_SIZE];
uint8_t m_Mask[IPV6_ADDR_SIZE];

Reply:
The macro `IPV6_ADDR_SIZE` is defined in `IpAddress.cpp`, I prefer not to move it to `IpAddress.h` because this header file is part of PcapPlusPlus API. I think it's better to just keep these values as 16